### PR TITLE
Sm/registry-completeness-bug

### PR DESCRIPTION
**Pull request by mshanemc**
### What does this PR do?
refactor some of the registry scripting stuff
fix a test that should not have been passing (wrong metadata coverage report)
add new 21 types that this unblocked

### What issues does this PR fix or reference?
@W-10758273@

### CODE REVIEW

File: src/registry/coverage.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  return (
+    JSON.parse((await got('https://mdcoverage.secure.force.com/services/apexrest/report')).body) as {
+      versions: { selected: number };
+    }
+  ).versions.selected;
+};
+
+export const getCoverage = async (apiVersion: number): Promise<CoverageObject> =>
+  JSON.parse(
+    (await got(`https://na${apiVersion - 9}.test1.pc-rnd.salesforce.com/mdcoverage/api.jsp`)).body
```
Comment (iowillhoit) [inline src/registry/coverage.ts:20]: Why `- 9`? Is this how far back we officially support? Might be worth adding a comment in the code here.

**Comment by iowillhoit** [comment]
- ✅  Reviewed code
- 🆗  Ran `yarn metadata:preview`- Output displayed for `v55` but confusing Slack Webhook Error. Could used a better message
- ✅ `yarn test:registry` passed 
- ✅ `yarn update-registry` ran without any new entries
- 🆗 `yarn update-supported-metadata` added one new entry (`MarketingAppExtAction`).

File: src/registry/coverage.ts (hunk 1 above)
Comment (mshanemc) [reply src/registry/coverage.ts:20]: good call.  I added a comment.

**Review by iowillhoit — APPROVED** [review]

File: src/registry/coverage.ts (hunk 1 above)
Comment (RupertBarrow) [reply src/registry/coverage.ts:22]: Where can I find this comment ?

Comment (RupertBarrow) [reply src/registry/coverage.ts:22]: Where can I find this comment @mshanemc ?

**Pull request merged**